src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -123,7 +123,46 @@ public BitArray(bool[] values)
             m_array = new int[GetInt32ArrayLengthFromBitLength(values.Length)];
             m_length = values.Length;
 
-            for (int i = 0; i < values.Length; i++)
+            int i = 0;
+
+            if (values.Length < Vector256<byte>.Count)
+            {
+                goto LessThan32;
+            }
+
+            if (Avx2.IsSupported)
+            {
+                fixed (bool* ptr = values)
+                {
+                    for (; (i + Vector256<byte>.Count) <= values.Length; i += Vector256<byte>.Count)
+                    {
+                        Vector256<byte> vector = Avx.LoadVector256((byte*)ptr + i);
+                        Vector256<byte> isFalse = Avx2.CompareEqual(vector, Vector256<byte>.Zero);
+                        int result = Avx2.MoveMask(isFalse);
+                        m_array[i / 32] = ~result;
```

Comment:
I guess in theory you can accumulate results in a Vector256 and then flush it to `m_array`

Reply:
>Also, you won't have to do ~ if you compare vector against `Vector256.Create((byte)1)`

This was done so that we could read `true` for any non-zero bytes.